corpus/segments.py

```diff
@@ -261,6 +262,43 @@ def run(self):
                 f.writelines(lines[start:end])
 
 
+class DynamicSplitSegmentFileJob(Job):
+
+    """
+    This split the segments to concurrent many shares.
+    It is a variant to the existing SplitSegmentFileJob, which requires a delayed variable for argument concurrent.
+    """
+
+    def __init__(self, segment_file: tk.Union[tk.Path, str], concurrent: tk.Delayed):
+        """
+        :param tk.Union[tk.Path, str] segment_file: segment file
+        :param tk.Delayed concurrent: number of splits
+        """
+        self.segment_file = segment_file
+        self.concurrent = concurrent
+        self.out_split_dir = self.output_path("split", directory=True)
+
+    def tasks(self):
+        yield Task("run", resume="run", mini_task=True)
+
+    def run(self):
+        with open(tk.uncached_path(self.segment_file), "rt") as f:
+            lines = [l for l in f.readlines() if len(l.strip()) > 0]
+
+        nb_seg = len(lines)
+        if isinstance(self.concurrent, tk.Delayed):
+            self.concurrent = self.concurrent.get()
+        seg_per_split = nb_seg // self.concurrent
+        nb_rest_seg = nb_seg % self.concurrent
+        end = 0
+        for i in range(1, self.concurrent + 1):
+            start = end
+            fpath = '{}/segments.{}'.format(self.out_split_dir, i)
```

Comment:
`self.out_split_dir` is a `tk.Path` object. Does it work with string `format`?

Reply:
yeah it works.

Reply:
This does work, but I am not sure if `__str__` returns the correct thing here (maybe this calls the caching...)

Reply:
ok then this need to be verified